cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -769,7 +770,7 @@ private SearchResults searchByCustomIndex(SearchRequest request) throws BadReque
           if (!metadataEntry.isPresent()) {
```

Comment:
You can remove this `if` since you are using the `optional.ifPresent` below.

Reply:
ok